File: app/src/main/java/io/github/domi04151309/powerapp/MainActivity.java
package io.github.domi04151309.powerapp;

import android.content.Intent;
import android.os.Build;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.KeyEvent;
import android.view.View;
import android.widget.Toast;

import java.io.DataOutputStream;
import java.io.IOException;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        Theme.check(this);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.shutdown).setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                try {
                    Process proc = Runtime.getRuntime()
                            .exec(new String[]{ "su", "-c", "reboot -p" });
                    proc.waitFor();
                } catch (Exception ex) {
                    Toast.makeText
                            (MainActivity.this, "Action failed! Please enable root!", Toast.LENGTH_LONG).show();
                }
            }
        });
        findViewById(R.id.reboot).setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                try {
                    Process proc = Runtime.getRuntime()
                            .exec(new String[]{ "su", "-c", "reboot" });
                    proc.waitFor();
                } catch (Exception ex) {
                    Toast.makeText(MainActivity.this, "Action failed! Please enable root!",
                            Toast.LENGTH_LONG).show();
                }
            }
        });
        findViewById(R.id.recovery).setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                try {
                    Process proc = Runtime.getRuntime()
                            .exec(new String[]{ "su", "-c", "reboot recovery" });
                    proc.waitFor();
                } catch (Exception ex) {
                    Toast.makeText(MainActivity.this, "Action failed! Please enable root!",
                            Toast.LENGTH_LONG).show();
                }
            }
        });
        findViewById(R.id.bootloader).setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                try {
                    Process proc = Runtime.getRuntime()
                            .exec(new String[]{ "su", "-c", "reboot bootloader" });
                    proc.waitFor();
                } catch (Exception ex) {
                    Toast.makeText(MainActivity.this, "Action failed! Please enable root!",
                            Toast.LENGTH_LONG).show();
                }
            }
        });
        findViewById(R.id.soft_reboot).setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                try {
                    Process proc = Runtime.getRuntime()
                            .exec(new String[]{ "su", "-c", "setprop ctl.restart zygote" });
                    proc.waitFor();
                } catch (Exception ex) {
                    Toast.makeText
                            (MainActivity.this, "Action failed! Please enable root!", Toast.LENGTH_LONG).show();
                }
            }
        });
        findViewById(R.id.system_ui).setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                try {
                    Process proc = Runtime.getRuntime()
                            .exec(new String[]{ "su", "-c", "killall com.android.systemui" });
                    proc.waitFor();
                } catch (Exception ex) {
                    Toast.makeText
                            (MainActivity.this, "Action failed! Please enable root!", Toast.LENGTH_LONG).show();
                }
            }
        });
        findViewById(R.id.screen_off).setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                try {
                    Process proc = Runtime.getRuntime()
                            .exec(new String[]{ "su", "-c", "input keyevent KEYCODE_POWER" });
                    proc.waitFor();
                } catch (Exception ex) {
                    Toast.makeText(MainActivity.this, "Action failed! Please enable root!",
                            Toast.LENGTH_LONG).show();
                }
            }
        });
        findViewById(R.id.root).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Process p;
                try {
                    p = Runtime.getRuntime().exec("su");
                    DataOutputStream os = new DataOutputStream(p.getOutputStream());
                    os.writeBytes("echo access granted\n");
                    os.writeBytes("exit\n");
                    os.flush();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        });
        findViewById(R.id.prefBtn).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, Preferences.class));
            }
        });
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event)  {
        if (keyCode == KeyEvent.KEYCODE_BACK) {
            onBackPressed();
            return true;
        }
        return super.onKeyDown(keyCode, event);
    }


    @Override
    public void onBackPressed() {
        startActivity(new Intent(Intent.ACTION_MAIN).addCategory(Intent.CATEGORY_HOME).setFlags(Intent.FLAG_ACTIVITY_NEW_TASK));
    }
}

File: app/src/main/java/io/github/domi04151309/powerapp/MainActivity.kt
package io.github.domi04151309.powerapp

import android.content.Intent
import androidx.appcompat.app.ActionBar
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.KeyEvent
import android.view.View

import java.io.DataOutputStream
import java.io.IOException

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        Theme.check(this)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if (Theme.customActionBar) {
            val actionBar = supportActionBar!!
            actionBar.displayOptions = ActionBar.DISPLAY_SHOW_CUSTOM
            actionBar.setDisplayShowCustomEnabled(true)
            actionBar.setCustomView(R.layout.action_bar)
            actionBar.elevation = 0f
            val scrollView = findViewById<View>(R.id.scrollView)
            scrollView.viewTreeObserver.addOnScrollChangedListener {
                if (scrollView.scrollY > 0)
                    actionBar.elevation = 16f
                else
                    actionBar.elevation = 0f
            }
        }

        val po = PowerOptions(this)

        findViewById<View>(R.id.shutdown).setOnClickListener {
            po.shutdown()
        }
        findViewById<View>(R.id.reboot).setOnClickListener {
            po.reboot()
        }
        findViewById<View>(R.id.recovery).setOnClickListener {
            po.rebootIntoRecovery()
        }
        findViewById<View>(R.id.bootloader).setOnClickListener {
            po.rebootIntoBootloader()
        }
        findViewById<View>(R.id.soft_reboot).setOnClickListener {
            po.softReboot()
        }
        findViewById<View>(R.id.system_ui).setOnClickListener {
            po.restartSystemUI()
        }
        findViewById<View>(R.id.screen_off).setOnClickListener {
            po.turnOffScreen()
        }
        findViewById<View>(R.id.root).setOnClickListener {
            val p: Process
            try {
                p = Runtime.getRuntime().exec("su")
                val os = DataOutputStream(p.outputStream)
                os.writeBytes("echo access granted\n")
                os.writeBytes("exit\n")
                os.flush()
            } catch (e: IOException) {
                e.printStackTrace()
            }
        }
        findViewById<View>(R.id.prefBtn).setOnClickListener { startActivity(Intent(this@MainActivity, Preferences::class.java)) }
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        if (keyCode == KeyEvent.KEYCODE_BACK) {
            onBackPressed()
            return true
        }
        return super.onKeyDown(keyCode, event)
    }

    override fun onBackPressed() {
        startActivity(Intent(Intent.ACTION_MAIN).addCategory(Intent.CATEGORY_HOME).setFlags(Intent.FLAG_ACTIVITY_NEW_TASK))
    }
}
